### PR TITLE
[Bot] Implement from issue #97: Create simple math utility with basic operations

### LINKED ISSUE #97: Create simple math utility with basic operations
﻿## Project Tag
project: math

## Description
Create a lightweight math utility module that provides basic arithmetic operations for use in other projects.

## Acceptance Criteria
- Function to add two numbers
- Function to multiply two numbers  
- Basic input validation
- Simple test to verify functions work
- Documentation with usage example

## Files to create
- \`math_utils.py\` - main utility functions
- \`test_math_utils.py\` - basic tests
- \`README.md\` - usage documentation

## Dependencies
- No external dependencies
- Use Python standard library only

## Scope & Segmentation Guidelines
**IMPORTANT FOR ANALYZER**: This is a simple, focused project. 
- Create maximum 3-4 logical tasks
- Group related activities together (don't create separate tasks for each function)
- Suggested segmentation: (1) Core implementation, (2) Testing, (3) Documentation
- Avoid micro-tasks like 'create __init__.py' or 'add single function'
- Keep tasks substantial and meaningful

This is a workflow verification test - prioritize logical task grouping over granular breakdown."

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #97

Project root: `projects/math`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-13 19:08:17 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of the math utility is a good start, but it lacks a division operation and has incomplete test coverage. Addressing these issues will improve functionality and reliability.

Files outside project root `projects`: README.md

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math/math_utils.py:15`**: The module does not handle division by zero.
  *Suggestion*: Implement a divide function that checks for division by zero and raises an appropriate exception.
- **`projects`**: Files modified outside enforced project root `projects`
  *Suggestion*: Move all changes under `projects`. Offending files: README.md

#### 💡 SUGGESTION
- **`projects/math/test_math_utils.py:9`**: Test cases for the multiply function are incomplete.
  *Suggestion*: Add more test cases for the multiply function, including edge cases like multiplying by zero and negative numbers.
- **`projects/math/math_utils.py`**: Consider adding a divide function to complete the basic arithmetic operations.
  *Suggestion*: Implement a divide function that performs division and handles division by zero.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math/math_utils.py
+++ b/projects/math/math_utils.py
@@ -0,0 +1,20 @@
+def divide(a, b):
+    """Return the result of dividing a by b."""
+    if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
+        raise ValueError("Both arguments must be numbers.")
+    if b == 0:
+        raise ValueError("Cannot divide by zero.")
+    return a / b
+
+def add(a, b):
+    """Return the sum of two numbers."""
+    if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
+        raise ValueError("Both arguments must be numbers.")
+    return a + b
+
+def multiply(a, b):
+    """Return the product of two numbers."""
+    if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
+        raise ValueError("Both arguments must be numbers.")
+    return a * b
+
 if __name__ == "__main__":
     print("Math Utility Module")
     print("Example: add(2, 3) =", add(2, 3))
     print("Example: multiply(2, 3) =", multiply(2, 3))
+    print("Example: divide(6, 3) =", divide(6, 3))
+    print("Example: divide(6, 0) =", divide(6, 0))
```

**Patch 2**

```diff
--- a/projects/math/test_math_utils.py
+++ b/projects/math/test_math_utils.py
@@ -0,0 +1,14 @@
 import unittest
 from math_utils import add, multiply, divide
 
 class TestMathUtils(unittest.TestCase):
     def test_add(self):
         self.assertEqual(add(2, 3), 5)
         self.assertRaises(ValueError, add, 2, "three")
 
     def test_multiply(self):
         self.assertEqual(multiply(2, 3), 6)
+        self.assertEqual(multiply(0, 5), 0)
+        self.assertEqual(multiply(-2, 3), -6)
+        self.assertRaises(ValueError, multiply, 2, "three")
+
+    def test_divide(self):
+        self.assertEqual(divide(6, 3), 2)
+        self.assertRaises(ValueError, divide, 6, 0)
+        self.assertRaises(ValueError, divide, 6, "three")
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!
